vulnfeeds/cves/versions.go

```diff
@@ -256,3 +272,29 @@ func CPEs(cve CVEItem) []string {
 	}
 	return cpes
 }
+
+// Parse a well-formed CPE string into a struct.
+func ParseCPE(cpe_str string) (*CPE, error) {
+	if !strings.HasPrefix(cpe_str, "cpe:") {
+		return nil, errors.New("Invalid CPE string")
+	}
+	if !strings.Contains(cpe_str, ":") {
+		return nil, errors.New("Invalid CPE string")
+	}
+
+	cpe_fields := strings.Split(cpe_str, ":")
```

Comment:
Add a check for length of `cpe_fields` and return an error if it doesn't match up?

Reply:
Good idea, thanks.